ydb/core/tx/datashard/prefix_kmeans.cpp

```diff
@@ -619,35 +686,28 @@ class TPrefixKMeansScan final: public TPrefixKMeansScanBase, private TCalculatio
             std::push_heap(MaxRows.begin(), MaxRows.end());
             MaxProbability = MaxRows.front().P;
         }
-        return MaxProbability != 0 ? EScan::Feed : EScan::Reset;
     }
 
-    EScan FeedKMeans(const TRow& row)
+    void FeedKMeans(TArrayRef<const TCell> row)
     {
-        Y_ASSERT(row.Size() == 1);
-        const ui32 pos = FeedEmbedding(*this, Clusters, row, 0);
-        AggregateToCluster(pos, row.Get(0).Data());
-        return EScan::Feed;
+        const ui32 pos = FeedEmbedding(*this, Clusters, row, EmbeddingPos);
+        AggregateToCluster(pos, row.at(EmbeddingPos).Data());
     }
 
-    EScan FeedUploadBuild2Build(TArrayRef<const TCell> key, const TRow& row)
+    void FeedUploadBuild2Build(TArrayRef<const TCell> key, TArrayRef<const TCell> row)
     {
         const ui32 pos = FeedEmbedding(*this, Clusters, row, EmbeddingPos);
-        if (pos > K) {
-            return EScan::Feed;
+        if (pos < K) {
+            AddRowBuild2Build(PostingBuf, Child + pos, key, row, PrefixColumns);
         }
-        AddRowBuild2Build(PostingBuf, Child + pos, key, row, PrefixColumns);
-        return FeedUpload();
     }
 
-    EScan FeedUploadBuild2Posting(TArrayRef<const TCell> key, const TRow& row)
+    void FeedUploadBuild2Posting(TArrayRef<const TCell> key, TArrayRef<const TCell> row)
     {
         const ui32 pos = FeedEmbedding(*this, Clusters, row, EmbeddingPos);
-        if (pos > K) {
-            return EScan::Feed;
+        if (pos < K) {
```

Comment:
Good catch, can you also fix same condition in other places (should be `>= K` instead of `> K`)?
It's not a bug, because only `[0...K)` and max_uint is possible, but still incorrect code